core/kernel/ts_manager.c

```diff
@@ -62,7 +62,13 @@ struct ts_session *ts_get_calling_session(void)
 
 struct ts_session *ts_get_current_session_may_fail(void)
 {
-	return TAILQ_FIRST(&thread_get_tsd()->sess_stack);
```

Comment:
Why doesn't this work any longer?

Reply:
that's right. Why is added function `thread_get_tsd_may_fail()` needed? Commit message says:
> Previously, ts_get_current_session_may_fail was calling
thread_get_tsd(), which may abort if current thread_id is invalid.

thread_id is a core information, unrelated to userland mapping. Is is really an issue in the scope of this P-R?